### PR TITLE
Feature/ Convert JS code to TS code

### PR DESCRIPTION
**Pull request by lokeshg-sde**

### CODE REVIEW

**Comment by lokeshg-sde** [comment]
@Vishwas-75  Fix Types in the Airline Controller and Middle ware auth.js then we can execute middle ware once in the app.

**Comment by lokeshg-sde** [comment]
and If possible add type declarations to the models

**Comment by lokeshg-sde** [comment]
@Vishwas-75  Merge this PR once you have tested, and fine with the code. this is working fine from my side with postmen.

File: server/types/api.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => unknown
+
+export type CallBackFunction = (req: Request, res: Response) => unknown
+
+export type Method = 'GET' | 'POST' | 'DELETE' | 'PATCH'
+
+export type UrlPath = { url: string; methods?: Method[] }
```
Comment (Vishwas-75) [inline server/types/api.ts]: Can we change the name from UrlPath to Routes??

File: server/types/api.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export type Paths = UrlPath[]
```
Comment (Vishwas-75) [inline server/types/api.ts]: I think we don't need this extra type we can use Routes[] || Array<Routes>

File: server/package-lock.json [hunk 3]
```diff
@@ -4,6 +4,33 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
```
Comment (Vishwas-75) [inline server/package-lock.json]: We can delete this lock json file any we don't need this one.

File: server/middleware/authentication.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import type { Request, Response, NextFunction } from 'express'
+
+import type { MiddlewareCallBackFunction, Paths, UrlPath, Method } from '../types/api'
+
+function matchCurrentUrl(paths: Paths, urlPath: string, method: string): boolean {
```
Comment (Vishwas-75) [inline server/middleware/authentication.ts]: can we change  **paths to privateRoutes**  and  **urlPath to route** for better understanding??

File: server/types/api.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+export type MiddlewareCallBackFunction = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => unknown;
+
+export type CallBackFunction = (req: Request, res: Response) => unknown;
+
+export type Method = "GET" | "POST" | "DELETE" | "PATCH";
```
Comment (Vishwas-75) [inline server/types/api.ts:11]: We can use an enum type for method since it has only 4 type

File: server/types/api.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+export type UserRole = "admin" | "owner" | "user";
```
Comment (Vishwas-75) [inline server/types/api.ts:12]: Here as well.

File: server/types/api.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+export type RouteInfo = {
+  url: string;
+  methods?: Method[];
```
Comment (Vishwas-75) [inline server/types/api.ts:16]: methods array??

**Comment by lokeshg-sde** [comment]
Let's merge this we can see those enum later, even we need to modify that authentication process as well.

File: server/types/api.ts (hunk 7 above)
Comment (lokeshg-sde) [reply server/types/api.ts:16]: Yes it will be so that we can control any type request of same URL with different method.

**Pull request merged**